urllib3/contrib/gaepool.py

```diff
@@ -0,0 +1,101 @@
+import warnings
+
+from ..connectionpool import (
+    ConnectionPool,
+    _Default,
+    get_host)
+
+from ..exceptions import (
+    HTTPWarning,
+    TimeoutError,
+)
+
+from ..packages.six import BytesIO
+from ..request import RequestMethods
+from ..response import HTTPResponse
+from ..util.timeout import Timeout
+from ..util.url import Url
+
+try:
+    from google.appengine.api import urlfetch
+except ImportError:
+    urlfetch = None
+
+
+class GAEPlatformWarning(HTTPWarning):
+    pass
+
+
+class GAEConnectionPool(ConnectionPool, RequestMethods):
+
+    def __init__(self, host, port=None, scheme='http', timeout=5, headers=None,
+                 retries=None, validate_certificate=False):
+
+        if not urlfetch:
+            raise EnvironmentError(
```

Comment:
`EnvironmentError` seems to be more for IO stuff. Maybe define our own `PlatformException` matching our `PlatformWarning`